pkg/client/unversioned/portforward/portforward.go

```diff
@@ -255,16 +256,20 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	errorChan := make(chan error)
 	doneChan := make(chan struct{}, 2)
 
+	connectionID := util.NewUUID()
+
 	// create error stream
 	headers := http.Header{}
 	headers.Set(api.StreamType, api.StreamTypeError)
 	headers.Set(api.PortHeader, fmt.Sprintf("%d", port.Remote))
+	headers.Set(api.ConnectionIDHeader, string(connectionID))
 	errorStream, err := pf.streamConn.CreateStream(headers)
 	if err != nil {
-		glog.Errorf("Error creating error stream for port %d -> %d: %v", port.Local, port.Remote, err)
+		util.HandleError(fmt.Errorf("error creating error stream for port %d -> %d: %v", port.Local, port.Remote, err))
 		return
 	}
-	defer errorStream.Reset()
+	// we're not writing to this stream
+	errorStream.Close()
 	go func() {
 		message, err := ioutil.ReadAll(errorStream)
```

Comment:
similar question to remoteCommand... should we always send to the errorChan (nil or an error) and unconditionally receive from errorChan before returning?